File: app/src/main/java/tourtle/ticketing2/utils/Session.java
package tourtle.ticketing2.utils;

import android.content.Context;
import android.content.SharedPreferences;

public class Session {

    private final SharedPreferences prefs;
    private final SharedPreferences.Editor editor;
    private String idWisata,namaWisata,idUser,footer,email;

    public Session(Context context) {
        prefs = context.getSharedPreferences("myapp", Context.MODE_PRIVATE);
        editor = prefs.edit();
        editor.apply();
    }

    public void setLoggedIn(boolean loggedin) {
        editor.putBoolean("loggedInmode", loggedin);
        editor.commit();
    }

    public void setIdWisata(String idWisata) {
        this.idWisata = idWisata;
        editor.putString("idWisata", this.idWisata);
        editor.commit();
    }

    public void setIdUser(String idUser) {
        this.idUser = idUser;
        editor.putString("idUser", this.idUser);
        editor.commit();
    }

    public void setEmail(String email) {
        this.email = email;
        editor.putString("email",this.email);
        editor.commit();
    }

    public void setNamaWisata(String namaWisata) {
        this.namaWisata = namaWisata;
        editor.putString("namaWisata", this.namaWisata);
        editor.commit();
    }


    public void setFooter(String footer) {
        this.footer = footer;
        editor.putString("footer",this.footer);
        editor.commit();
    }

    public String getIdWisata(){
        return prefs.getString("idWisata", this.idWisata);
    }


    public String getEmail() {
        return prefs.getString("email",this.email);
    }

    public boolean loggedin() {
        return prefs.getBoolean("loggedInmode", false);
    }

    public String getNamaWisata() {
        return prefs.getString("namaWisata", this.namaWisata);
    }

    public String getIdUser() {
        return prefs.getString("idUser",this.idUser);
    }

    public String getFooter() {
        return prefs.getString("footer",this.footer);
    }


}

File: app/src/main/java/tourtle/ticketing2/utils/Session.kt
package tourtle.ticketing2.utils

import android.content.Context
import android.content.SharedPreferences

class Session(context: Context?) {
    private val prefs: SharedPreferences?
    private val editor: SharedPreferences.Editor?
    private var idWisata: String? = null
    private var namaWisata: String? = null
    private var idUser: String? = null
    private var footer: String? = null
    private var email: String? = null

    fun setLoggedIn(loggedin: Boolean) {
        editor?.putBoolean("loggedInmode", loggedin)
        editor?.commit()
    }

    fun setIdWisata(idWisata: String?) {
        this.idWisata = idWisata
        editor?.putString("idWisata", this.idWisata)
        editor?.commit()
    }

    fun setIdUser(idUser: String?) {
        this.idUser = idUser
        editor?.putString("idUser", this.idUser)
        editor?.commit()
    }

    fun setEmail(email: String?) {
        this.email = email
        editor?.putString("email", this.email)
        editor?.commit()
    }

    fun setNamaWisata(namaWisata: String?) {
        this.namaWisata = namaWisata
        editor?.putString("namaWisata", this.namaWisata)
        editor?.commit()
    }

    fun setFooter(footer: String?) {
        this.footer = footer
        editor?.putString("footer", this.footer)
        editor?.commit()
    }

    fun getIdWisata(): String? {
        return prefs?.getString("idWisata", idWisata)
    }

    fun getEmail(): String? {
        return prefs?.getString("email", email)
    }

    fun loggedin(): Boolean {
        return prefs!!.getBoolean("loggedInmode", false)
    }

    fun getNamaWisata(): String? {
        return prefs?.getString("namaWisata", namaWisata)
    }

    fun getIdUser(): String? {
        return prefs?.getString("idUser", idUser)
    }

    fun getFooter(): String? {
        return prefs?.getString("footer", footer)
    }

    init {
        prefs = context?.getSharedPreferences("myapp", Context.MODE_PRIVATE)
        editor = prefs?.edit()
        editor?.apply()
    }
}